Mage.Sets/src/mage/cards/p/Prophecy.java

```diff
@@ -0,0 +1,89 @@
+package mage.cards.p;
+
+import java.util.UUID;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.common.delayed.AtTheBeginOfNextUpkeepDelayedTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateDelayedTriggeredAbilityEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.common.TargetOpponent;
+
+/**
+ *
+ * @author noahg
+ */
+public final class Prophecy extends CardImpl {
+
+    public Prophecy(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{W}");
+        
+
+        // Reveal the top card of target opponent's library. If it's a land, you gain 1 life. Then that player shuffles his or her library.
+        this.getSpellAbility().addEffect(new ProphecyEffect());
+        this.getSpellAbility().addTarget(new TargetOpponent());
+
+        // Draw a card at the beginning of the next turn's upkeep.
+        this.getSpellAbility().addEffect(new CreateDelayedTriggeredAbilityEffect(new AtTheBeginOfNextUpkeepDelayedTriggeredAbility(new DrawCardSourceControllerEffect(1)), false));
+    }
+
+    public Prophecy(final Prophecy card) {
+        super(card);
+    }
+
+    @Override
+    public Prophecy copy() {
+        return new Prophecy(this);
+    }
+}
+
+class ProphecyEffect extends OneShotEffect {
+
+    public ProphecyEffect() {
+        super(Outcome.GainLife);
+        this.staticText = "Reveal the top card of target opponent's library. If it's a land, you gain 1 life. Then that player shuffles his or her library";
```

Comment:
It should be "Then that player shuffles their library", the oracle text in the spoiler file is outdated. I've been dragging my feet on updating all of it.